awx/ui_next/src/screens/Template/shared/JobTemplateForm.jsx

```diff
@@ -36,22 +50,113 @@ class JobTemplateForm extends Component {
       playbook: '',
       summary_fields: {
         inventory: null,
+        labels: { results: [] },
       },
     },
   };
 
   constructor(props) {
     super(props);
-
     this.state = {
+      hasContentLoading: true,
+      contentError: false,
+      loadedLabels: [],
+      newLabels: [],
+      removedLabels: [],
       inventory: props.template.summary_fields.inventory,
     };
+    this.handleNewLabel = this.handleNewLabel.bind(this);
+    this.loadLabels = this.loadLabels.bind(this);
+    this.disassociateLabel = this.disassociateLabel.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadLabels(QSConfig);
+  }
+
+  async loadLabels(QueryConfig) {
+    this.setState({ contentError: null, hasContentLoading: true });
+    let loadedLabels;
+    try {
+      const { data } = await LabelsAPI.read(QueryConfig);
+      loadedLabels = [...data.results];
+      if (data.next && data.next.includes('page=2')) {
```

Comment:
The assumption we're making here is important - we're fetching up to the first 400 labels in the api. The implication here is that more than 400 labels isn't supported in the UI without a code update.

For what its worth I think this is totally fine, but we should document this with a comment or something.

Reply:
👍